lms/djangoapps/learner_home/serializers.py

```diff
@@ -427,8 +440,20 @@ class _PseudoSessionCourseSerializer(serializers.Serializer):
     enrollment = LiteralField(STATIC_ENTITLEMENT_ENROLLMENT_DATA)
 
     def get_course(self, instance):
-        pseudo_session = self.context['unfulfilled_entitlement_pseudo_sessions'].get(str(instance.uuid))
-        return UnfulfilledEntitlementSerializer._PseudoSessionCourseSerializer(pseudo_session).data
+        pseudo_session = self.context["unfulfilled_entitlement_pseudo_sessions"].get(
+            str(instance.uuid)
+        )
+        return UnfulfilledEntitlementSerializer._PseudoSessionCourseSerializer(
+            pseudo_session
+        ).data
+
+    def get_courseProvider(self, entitlement):
+        """Look up course provider from CourseOverview matching the pseudo session"""
+        pseudo_session = self.context["unfulfilled_entitlement_pseudo_sessions"].get(
+            str(entitlement.uuid)
+        )
+        course_overview = CourseOverview.get_from_id(pseudo_session["key"]) or None
```

Comment:
Suggestion: move this loading back into the view, prefetch these for unfulfilled entitlements and provide them via context. That we we can (best case) get all course overviews in one query

Reply:
Also, no test covering this